src/arizona_live_view.erl

```diff
@@ -36,9 +36,14 @@ Live view.
 -callback mount(Socket) -> Socket
     when Socket :: arizona_socket:t().
 
--callback render(Macros) -> Tree
-    when Macros :: arizona_template_compiler:macros(),
-         Tree :: arizona_tpl_parse:tree().
+-callback render(Macros1) -> Result
+    when Macros1 :: arizona_template_compiler:macros(),
+         Result :: {ok, {Parsed, Macros2}}
+                 | {error, ErrReason},
+         Parsed :: [arizona_template_parser:element()],
+         Macros2 :: arizona_template_compiler:macros(),
+         ErrReason :: arizona_template_scanner:error_reason()
+                    | arizona_template_parser:error_reason().
```

Comment:
Does this mean we'd not know if the error comes from one of the other, and force consumer to check? I think `{error, {scanner, Error}} | {error, {parser, Error}}` would be helpful (?)

Reply:
Sounds good. I'll change it to `{error, {scanner, Error}} | {error, {parser, Error}}`.

Reply:
[37c2b46](https://github.com/arizona-framework/arizona/pull/177/commits/37c2b46e002b0f6f819bf25369706f0e3b436912)